### ISSUE TITLE
BUG

### ISSUE THREAD

**Issue by Pelfa** [Dar9586/NClientV2#402]
Hello.
   I wanted to make this short and clarify the bug easily.. basically it keeps saying no entries found..
Althought i tried to see if its the websites problem its not.. also idk if it has been there before since its been long since i last visited the website but it has a DDoS Protection now
The app was working fine just hours ago. I tried fixing the problem by reinstalling/updating the app but the problem stays the same.. i resetted the app, restarted my phone.. doesnt work really.. so i hope u fix this bug... Or maybe time will fix it and my phone is just being dumb right now.
Thx!!

**Comment by Evand-smith97** [comment]
yeah, its happening in my phone too. I thought it was server down from the og site, yet the og site works fine.

**Comment by Notarin** [comment]
Hi, the title of your issue isn't descriptive at all of you issue, if every bug was named "BUG" nobody would be able to navigate the issues, I almost made my own issue instead of commenting here because I didn't even realize that this was the same issue I'm experiencing, from now on, when making issues, please title them accordingly. A good title would be  "No Entries Found"

**Comment by Notarin** [comment]
Also, I'm having the same issue as well, most likely either the API has been updated, or if the app scrapes, the site itself has been updated.

**Comment by Notarin** [comment]
This was also mentioned in #386

**Comment by Notarin** [comment]
If it uses the NHentai NPM API then that appears to be currently downhttps://github.com/Zekfad/nhentai-api/issues/24

**Comment by Kinuseka** [comment]
Seems like nhentai added a cloudflare system. So I can guess that this major system upgrade is also interfering with the API hence the downtime.

**Comment by Magnetogenerator** [comment]
Same issue, a nhentai link cant be opened by NClientV2, on my phone it will do nothing and directly go back to the former app, but it can be opened and load content well by a web browser after a cloudflare check, that is probably where the problem lies.

**Comment by Notarin** [comment]
After looking through the app, I've found that you can still open your favorites and even download them.

**Comment by Notarin** [comment]
I've found this online https://www.reddit.com/r/wholesomehentai/comments/t7gf2q/please_read_before_posting_an_nhentai_link/
![image](https://user-images.githubusercontent.com/25104390/156926530-11ae0d9a-e443-43e8-bd2d-a98fe67df295.png)

**Comment by Notarin** [comment]
I've received a response, They've never actually found a solution, they just don't scrape the site anymore.

**Comment by Kinuseka** [comment]
> I've received a response, They've never actually found a solution, they just don't scrape the site anymore.

Seems like they had a private API access. I wonder if they ever consider making it public

**Comment by Kinuseka** [comment]
I propose a solution; 
What I found is that the site does not need to reload cloudflare when a browser is deemed legitimate.
We can perhaps integrate a browser open it on first boot, and let the cloudflare do its job, and take the session and or cookie/token after the authentication with the integrated browser. 
The app will now load as normal. If the token/cookie expires we reload with the same technique.
I dont know if this abides ToS but the APP only rearranges contents of the website to make it compatible into the mobile viewing. Alternatively we can also wait for the new API to drop. Goodluck

**Comment by keatit71** [comment]
Another solution to cloud flare protection or anti-scraping i've seen is using this Program called Flaresolverr. https://github.com/FlareSolverr/FlareSolverr

**Comment by Kinuseka** [comment]
> Another solution to cloud flare protection or anti-scraping i've seen is using this Program called Flaresolverr. https://github.com/FlareSolverr/FlareSolverr

This is pretty much what I said. The existence of this repo might make it easier

**Comment by keatit71** [comment]
Flaresolverr is already used to bypass cloudflare restrictions on torrent sites for programs like jackett which is how i found it so it should work well in this regard.

**Comment by Notarin** [comment]
I'm not sure if anyone has noticed, but it appears that the issue has been solved, app is working normal for me

**Comment by Notarin** [comment]
> > I've received a response, They've never actually found a solution, they just don't scrape the site anymore.
> 
> Seems like they had a private API access. I wonder if they ever consider making it public

No, they literally just completely stopped scraping the site altogether
![image](https://user-images.githubusercontent.com/25104390/156932460-63488045-1d8a-49ef-b049-5d77787308ad.png)

**Comment by chayleaf** [comment]
https://github.com/Dar9586/NClientV2/pull/428 should fix this

**Comment by Notarin** [comment]
If this is fixed by #428 and it's now merged, shouldn't this be closed?
